cmd/build/command.go

```diff
@@ -80,16 +85,21 @@ func Build(ctx context.Context) *cobra.Command {
 	return cmd
 }
 
-func (bc *Command) getManifestAndBuildVersion(options *types.BuildOptions) (*model.Manifest, bool) {
+func (bc *Command) getManifestAndBuildVersion(options *types.BuildOptions) (*model.Manifest, bool, error) {
 	manifest, errManifest := bc.GetManifest(options.File)
-	if errManifest != nil {
-		oktetoLog.Debug("error getting manifest v2 from file %s: %v. Fallback to build v1", options.File, errManifest)
-	}
 
 	isBuildV2 := errManifest == nil &&
 		manifest.IsV2 &&
 		len(manifest.Build) != 0
-	return manifest, isBuildV2
+
+	if errManifest != nil {
+		if strings.HasPrefix(errManifest.Error(), "invalid manifest") {
```

Comment:
We should not be using the error string to perform any logic. This should be changed to use error types or specific error instances. This way we can rely on the type system and get the full benefit of Go.

Reply:
This means changing some of the `bc.GetManifest` code.

Reply:
`Read` function within `manifest.go` file is probably the place where we should return a specific error type to do what Jacob says.